### PR TITLE
Generate Thrift C++/JS/Java files into the build directory

### LINKED ISSUE #339: Move Thrift generated files into the build directory
Upon CodeCompass compilation, Thrift currently generates files into the source folder (`gen-cpp`, `gen-java`, `gen-js` directories), instead of the build location. Since these are generated files, they should be created in the build folder, which would also result in an easier cleanup.

### PR DESCRIPTION
**Pull request by mcserep**
Closes #339 .

### CODE REVIEW

**Comment by whisperity** [comment]
Partially supersedes #270.

**Comment by mcserep** [comment]
Can you review this @whisperity ? (@bruntib  is reviewing another PR.)  
Thanks.

File: .gitignore [hunk 1]
```diff
@@ -5,13 +5,3 @@ nbproject/
 *.swp
 build/
 
-# Generated files and directories.
-
-**/*odb\.cxx
```
Comment (whisperity) [inline .gitignore]: These lines for ODB remained here due to a rebase missing? This patch in itself did not touch ODB-related stuff.

Comment (mcserep) [reply .gitignore]: Yes, those should have been cleaned up in #323.

**Pull request merged**